zanata-war/src/main/java/org/zanata/rest/editor/service/resource/SuggestionsResource.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2015, Red Hat, Inc. and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.zanata.rest.editor.service.resource;
+
+import org.zanata.rest.editor.MediaTypes;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.List;
+
+/**
+ * Endpoint to search for suggestions from translation memory and other sources.
+ */
+@Produces({ MediaType.APPLICATION_JSON })
+@Consumes({ MediaType.APPLICATION_JSON })
+public interface SuggestionsResource {
+
+    public static final String SERVICE_PATH = "/suggestions/{sourceLocale}/{transLocale}";
```

Comment:
I would suggest moving the sourceLocale and transLocale from path params to query params. At least move transLocle because sourceLocale will most likely be en-US or en. Some locales with variable such as 'kw@kkcor' will look odd in path after encoding.